src/base.jl

```diff
@@ -271,16 +271,19 @@ Clear any value stored in ext.
 """
 clear_ext!(sys::System) = IS.clear_ext!(sys.internal)
 
-function set_unit_system!(component::Component, settings::SystemUnitsSettings)
-    component.internal.units_info = settings
+function set_units_setting!(
```

Comment:
@jd-lara Are you OK with the name change? If not, Sourabh could add a decrecated `set_unit_system! = set_units_setting!` function.

Reply:
This could be problematic since it will define the system unit system. I don't object but this is awkward

Reply:
I added a comment that these function names were unnecessarily not symmetric.
```
function set_unit_system!(component::Component, settings::SystemUnitsSettings)
function get_units_setting(component::T) where {T <: Component}
function has_units_setting(component::T) where {T <: Component}
```
Sourabh is making the names consistent and not changing any functionality. To your point about changing the system setting: that may or may not be true. In the case where it is true, we shouldn't allow it. That could be dangerous. That should not be exported to users. I'm really not liking this whole solution...